### PR TITLE
add Record.get default value

### PR DESCRIPTION
**Pull request by MykolaGolubyev**

### CODE REVIEW

File: webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java [hunk 1]
```diff
@@ -111,6 +111,16 @@ public String toString() {
...
     }
 
+    /**
+     * column index by column name
+     * @param columnName column name to get index for
+     * @return column index or -1 if not found
+     */
+    public int findColumnIdxByName(String columnName) {
+        Integer idx = indexByName.get(columnName);
+        return idx == null ? -1: idx;
```
Comment (tsiq-clemens) [inline webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java]: do we want to make `-1` a constant "COLUMN_NOT_FOUND" or some such?

File: webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java [hunk 2]
```diff
@@ -111,6 +111,16 @@ public String toString() {
         return namesByIndex.toString();
     }
 
+    /**
+     * column index by column name
+     * @param columnName column name to get index for
+     * @return column index or -1 if not found
+     */
+    public int findColumnIdxByName(String columnName) {
+        Integer idx = indexByName.get(columnName);
```
Comment (tsiq-clemens) [inline webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java]: out of curiosity: what does indexByName.get(null) return? null or exception?

File: webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy [hunk 3]
```diff
@@ -39,4 +39,24 @@ class RecordTest {
         def record = new Record(new Header(["c1", "c2", "c3"].stream()), ["v1", "v2", "v3"].stream())
         assert record.key == null
     }
+
+    @Test
+    void "should return defaultValue if a columnName is not present"() {
+        def record = new Record(new Header(["c1", "c2", "c3"].stream()), ["v1", "v2", "v3"].stream())
+        def valueC1 = record.get("c1", 42)
+        def valueC4 = record.get("c4", 42)
```
Comment (tsiq-clemens) [inline webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy]: or `get("noSuchColumn", 42)`?

File: webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy [hunk 4]
```diff
@@ -39,4 +39,24 @@ class RecordTest {
...
+        assert valueC1 == "v1"
+        assert valueC4 == 42
+    }
+
+    @Test
+    void "should return defaultValue if an idx is not present"() {
+        def record = new Record(new Header(["c1", "c2", "c3"].stream()), ["v1", "v2", "v3"].stream())
+        def valueC2 = record.get(1, 42)
+        def valueCN = record.get(5, 42)
```
Comment (tsiq-clemens) [inline webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy]: why 5? 3 is the edge case. I'd also indicate the too-high-index in the variable name

File: webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java (hunk 1 above)
Comment (MykolaGolubyev) [reply webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java]: Not to keen on that. I think by now -1 is a common knowledge.

File: webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy (hunk 3 above)
Comment (MykolaGolubyev) [reply webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy]: Intention here is to show more like columns that can be there but not in this case. I.e optional table columns. But I see what you mean. Sometimes I do that.

File: webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy (hunk 4 above)
Comment (MykolaGolubyev) [reply webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy]: 5 cause more than 2 :)
I feel like the scope of this test is to small to introduce named vars. If you feel strongly about it, I will change.

File: webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java (hunk 2 above)
Comment (MykolaGolubyev) [reply webtau-core/src/main/java/com/twosigma/webtau/data/table/header/Header.java]: null

File: webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy (hunk 4 above)
Comment (MykolaGolubyev) [reply webtau-core/src/test/groovy/com/twosigma/webtau/data/table/RecordTest.groovy]: Ok, changed :)
